exchange/replication.go

```diff
@@ -347,11 +350,21 @@ func (r *Replication) handleRequest(s network.Stream) {
 		// Create a new store to receive our new blocks
 		// It will be automatically picked up in the TransportConfigurer
 		storeID := r.idx.ms.Next()
-		err = r.idx.SetRef(&DataRef{
+
+		ref := &DataRef{
 			PayloadCID:  req.PayloadCID,
 			PayloadSize: int64(req.Size),
 			StoreID:     storeID,
-		})
+			Keys:        [][]byte{},
+		}
+
+		r.idx.GetRef(req.PayloadCID)
+
+		//for key := range tx.entries {
+		//	ref.Keys[key] = true
+		//}
+
+		err = r.idx.SetRef(ref)
```

Comment:
@gallexis l371 is where I was thinking you could iterate directly on channel states based on the channel ID, so you can get the keys and set them on the ref directly. Then you can remove the event subscribe in the NewReplication constructor. Makes sense?

Reply:
Ah yes indeed it makes more sense like that 👍